torch/csrc/jit/codegen/cuda/lower_utils.cpp

```diff
@@ -168,6 +168,29 @@ ParallelTypeBitmap getParallelBroadcastDomains(
   return parallel_broadcast & preds.at(tv).pred;
 }
 
+kir::Expr* applyReplacements(
```

Comment:
Can you please comment what this function is supposed to do? Initially, I thought applying replacements to `expr` would return the `expr` with some replacements inside the expression, but that's not the case when there is a mapping of the expression in the replacement map. If not mapped and the expression has scopes, it does return the original `expr` with replacements applied. Even in that case, would it be safe to just assign the return value of `applyReplacements` to `scope[i]`? Wouldn't it be necessary to do `Expr::setScope(scope)` as well? See, for example, `Scope::push_back(Expr*)`. 

Manipulating KIR isn't particularly intuitive at this point, and things like this function would be definitely something useful. Maybe @tlemo has something to suggest?

Reply:
OK, I realized this was just moved out of lower_unroll. Now that it sits in a util namespace, it needs some documentation.

Reply:
👍🏼